### DISCUSSION TITLE
Global WinUI3 application problem with MenuFlyout

### DISCUSSION THREAD

**Discussion by JozefGula** [microsoft/WindowsAppSDK#3899] [Q&A] [answered]
Hello, I am completely desperate and therefore I would like to ask if anyone has the same experience or an idea how to solve this problem:

Project file:

```
<PropertyGroup>
	<OutputType>WinExe</OutputType>
	<TargetFramework>net7.0-windows10.0.19041.0</TargetFramework>
	<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
	<RootNamespace>MyApp</RootNamespace>
	<ApplicationManifest>app.manifest</ApplicationManifest>
	<Platforms>x86;x64;ARM64</Platforms>
	<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
	<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
	<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
	<UseWinUI>true</UseWinUI>
	<Nullable>enable</Nullable>
	<ApplicationIcon>Assets\Icons\App.ico</ApplicationIcon>
	<EnableMsixTooling>true</EnableMsixTooling>
	<AssemblyVersion>1.0.23274.0926</AssemblyVersion>
	<BuiltInComInteropSupport>true</BuiltInComInteropSupport>
	<FileVersion>1.0.23274.0926</FileVersion>
</PropertyGroup>
```

Packages:

```
<ItemGroup>
	<PackageReference Include="AutoMapper" Version="12.0.1" />
	<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.1" />
	<PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
	<PackageReference Include="CommunityToolkit.WinUI.UI.Controls.DataGrid" Version="7.1.2" />
	<PackageReference Include="CommunityToolkit.WinUI.UI.Media" Version="7.1.2" />
	<PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.11" />
	<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="7.0.11">
		<PrivateAssets>all</PrivateAssets>
	</PackageReference>
	<PackageReference Include="Microsoft.Extensions.Hosting" Version="7.0.1" />
	<PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="7.0.0" />
	<PackageReference Include="Microsoft.Windows.CsWin32" Version="0.3.46-beta">
		<PrivateAssets>all</PrivateAssets>
		<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
	</PackageReference>
	<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.4.230913002" />
	<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
	<PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
	<PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
	<PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.11" />
	<PackageReference Include="Nucs.JsonSettings" Version="2.0.2" />
	<PackageReference Include="Serilog" Version="3.0.1" />
	<PackageReference Include="Serilog.Expressions" Version="3.4.1" />
	<PackageReference Include="Serilog.Extensions.Logging.File" Version="3.0.0" />
	<PackageReference Include="Serilog.Settings.Configuration" Version="7.0.1" />
	<PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
	<PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
	<PackageReference Include="Serilog.Sinks.File.Archive" Version="1.0.3" />
	<PackageReference Include="WinUICommunity.Components" Version="5.2.0" />
	<PackageReference Include="WinUICommunity.Core" Version="5.2.0" />
	<Manifest Include="$(ApplicationManifest)" />
</ItemGroup>
```

I created an application and I use MenuFlyout in it. At the beginning of the development of the project, MenuFlyout worked correctly, but I gradually added new and new packages to the project. I added migration for SQlite, divided the project into 3 libraries, etc. Now I am back to the test and I find that the MenuFlyout doesn't work for me in the whole application. After calling "ShowAt" with any parameters, the whole application crashes. Try/Catch does not catch any exception and the only thing I can find is this in the Event Viewer:

```
Faulting application name: MyApp.exe, version: 1.0.23274.926, time stamp: 0x64e80000
Faulting module name: KERNELBASE.dll, version: 10.0.19041.3393, time stamp: 0x6b4de7c9
Exception code: 0xc000027b
Fault offset: 0x000000000012d9b2
Faulting process id: 0x3b98
Faulting application start time: 0x01d9f78c4746f838
Faulting application path: E:\Software\MyApp\MyApp\bin\x64\Debug\net7.0-windows10.0.19041.0\MyApp.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 06ccb4a2-5e19-47ee-8450-cd490f998ff6
Faulting package full name: 
Faulting package-relative application ID: 
```

Unfortunately, I am making this application without git, so I can not just go back to the past and some specific point. However, since it happens in the entire application, the only thing I can think of is that it is related to some package.

Is there any way to find out what is happening in KERNELBASE or what kind of error really occurs there? Any other ideas?

After changing the debugger configuration I was able to get to this:
![image](https://github.com/microsoft/WindowsAppSDK/assets/8478191/edce844a-59d4-456c-bfca-030c33c22864)

Thank you very much

**Comment by JozefGula** [comment]
Ok, I finally found out that one line in my project causes this case:

```
<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
```

1. MenuFlyout works without this line in the project, but it does not work with this line in my project
2. My app does not work after publishing without this line in the project, but it works with this line in the project

These two things are mutually exclusive and I need both the application and the MenuFlyout to work. However, I have no idea how to proceed further...

Any ideas?

Thank you very much

**Comment by JozefGula** [comment]
I created a bug based on my findings: https://github.com/microsoft/microsoft-ui-xaml/issues/8954

**Comment by ghost1372** [comment]
i have menuflyout and working fine, you can test WinUI 3 Gallery app.
its unclear how you defined menuflyout, mybe issue is coming from there.
if you could not find anything, so i think you should wait for v1.4.2 which symbols will be available and you can debug easily

**Reply by JozefGula** [reply, marked as answer]
@ghost1372 thank you for your reply. It seems that my problem was caused by the ".vs" folder in the project. When I deleted this folder, the problem was solved.

I've covered more here: https://github.com/microsoft/microsoft-ui-xaml/issues/8954

Thank you very much
